lib/topological_inventory/core/service_catalog_client.rb

```diff
@@ -0,0 +1,53 @@
+require "rest_client"
+require "topological_inventory/core/service_plan_client"
+
+module TopologicalInventory
+  module Core
+    class ServiceCatalogClient
+      def initialize(source)
+        @source = source
+      end
+
+      # TODO? Move this to a module in the API repo
+      def order_service_plan(plan_name, service_offering_name, additional_parameters)
+        payload = ServicePlanClient.new.build_payload(
+          plan_name, service_offering_name, catalog_parameters(additional_parameters)
+        )
+        response = make_request(:post, order_service_plan_url, payload)
+        JSON.parse(response.body)
+      end
+
+      private
+
+      def order_service_plan_url
+        URI.join(@source.default_endpoint.base_url_path, "apis/servicecatalog.k8s.io/v1beta1/namespaces/default/serviceinstances").to_s
```

Comment:
> `namespaces/default/`

Hm the target namespace should be a parameter right @mkanoor?  Hard-coding to the default-namespace looks like a problem

Reply:
@agrare 
We are passing the namespace as part of the provider control parameters. The UI is going to have to drop down to pick the namespace.

Reply:
https://github.com/ManageIQ/service_portal-api/blob/60965b031a61570ca8adf0b0ee6a4c025ca12faa/public/doc/swagger-2.yaml#L526